lib/grape/middleware/error.rb

```diff
@@ -45,6 +45,10 @@ def find_handler(klass)
         handler = options[:rescue_handlers].find(-> { [] }) { |error, _| klass <= error }[1]
         handler ||= options[:base_only_rescue_handlers][klass]
         handler ||= options[:all_rescue_handler]
+        with_option = options[:rescue_options][:with]
+        if with_option.instance_of?(Symbol) && respond_to?(with_option)
```

Comment:
I see. I vote for the simplest and least error prone implementation and for catching bugs "early".